File: backend/src/main/java/com/smartcampus/backend/service/BookingService.java
package com.smartcampus.backend.service;

import com.smartcampus.backend.dto.BookingRequestDTO;
import com.smartcampus.backend.dto.BookingResponseDTO;
import com.smartcampus.backend.model.BookingStatus;

import java.util.List;

public interface BookingService {

    BookingResponseDTO createBooking(BookingRequestDTO request);

    BookingResponseDTO getBookingById(String id);

    List<BookingResponseDTO> getAllBookings();

    List<BookingResponseDTO> getBookingsByUserId(String userId);

    List<BookingResponseDTO> getBookingsByStatus(BookingStatus status);

    BookingResponseDTO updateBooking(String id, BookingRequestDTO request);

    BookingResponseDTO updateBookingStatus(String id, BookingStatus status);

    void deleteBooking(String id);
}

File: frontend/PAF/src/services/bookingService.js
import api from '../utils/api';

const BOOKING_URL = '/bookings';

const bookingService = {
  createBooking: async (bookingData) => {
    const response = await api.post(BOOKING_URL, bookingData);
    return response.data;
  },

  getBookingById: async (id) => {
    const response = await api.get(`${BOOKING_URL}/${id}`);
    return response.data;
  },

  getAllBookings: async () => {
    const response = await api.get(BOOKING_URL);
    return response.data;
  },

  getBookingsByUserId: async (userId) => {
    const response = await api.get(`${BOOKING_URL}/user/${userId}`);
    return response.data;
  },

  getBookingsByStatus: async (status) => {
    const response = await api.get(`${BOOKING_URL}/status/${status}`);
    return response.data;
  },

  updateBooking: async (id, bookingData) => {
    const response = await api.put(`${BOOKING_URL}/${id}`, bookingData);
    return response.data;
  },

  updateBookingStatus: async (id, status) => {
    const response = await api.patch(`${BOOKING_URL}/${id}/status`, { status });
    return response.data;
  },

  deleteBooking: async (id) => {
    await api.delete(`${BOOKING_URL}/${id}`);
  },
};

export default bookingService;
